portal/src/components/leaderboard/LeaderboardAdminView.tsx

```diff
@@ -0,0 +1,315 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { useRouter } from "next/router";
+
+import React, { useContext, useEffect, useState } from "react";
+import { useQuery } from "react-query";
+
+import {
+  Box,
+  Heading,
+  Flex,
+  Text,
+  Button,
+  Input,
+  Spinner,
+  HStack,
+  GridItem,
+  UnorderedList,
+  ListItem,
+  useMediaQuery,
+  Spacer,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  useDisclosure,
+} from "@chakra-ui/react";
+
+import { AddIcon } from "@chakra-ui/icons";
+import { FiEdit2 } from "react-icons/fi";
+
+import Web3 from "web3";
+
+import http from "../../utils/httpMoonstream";
+import queryCacheProps from "../../hooks/hookCommon";
+import Web3Context from "../../contexts/Web3Context/context";
+import LeaderboardMetadata from "./LeaderboardMetadata";
+import NewLeaderboard from "./NewLeaderboard";
+import EditLeaderboard from "./EditLeaderboard";
+import LeaderboardAPI from "./LeaderboardAPI";
+
+import ReactMarkdown from "react-markdown";
+import remarkGfm from "remark-gfm";
+import useMoonToast from "../../hooks/useMoonToast";
+import LeaderboardUpload from "./LeaderboardUpload";
+import { Score } from "./types";
+
+const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+const LeaderboardAdminView = () => {
+  const router = useRouter();
+  const toast = useMoonToast();
+  const [currentAccount, setCurrentAccount] = React.useState(ZERO_ADDRESS);
+  const web3ctx = useContext(Web3Context);
+  const [selectedIndex, setSelectedIndex] = React.useState(0);
+  const [status, setStatus] = React.useState("normal");
+  const [isUploading, setIsUploading] = useState(false);
+  const [title, setTitle] = React.useState("");
+  const [description, setDescription] = React.useState("");
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const [isVerySmallScreen, isSmallScreen] = useMediaQuery([
+    "(max-width: 440px)",
+    "(max-width: 810px)",
+  ]);
+
+  const getLeadeerboards = () => {
+    return http({
+      method: "GET",
+      url: "https://engineapi.moonstream.to/leaderboard/leaderboards",
+    }).then((res: any) => {
+      res.data.sort((a: any, b: any) => {
+        return b.created_at > a.created_at;
+      });
+      return res.data;
+    });
+  };
+
+  const leaderboardsQuery = useQuery(["leaderboards"], getLeadeerboards, {
+    onSuccess: (data) => {
+      console.log(data);
+    },
+    onError: (e) => {
+      console.log(e);
+    },
+  });
+
+  const getLeaderboardInfo = (id: string) => {
+    return http({
+      method: "GET",
+      url: `https://engineapi.moonstream.to/leaderboard/info?leaderboard_id=${id}`,
+    }).then((res: any) => {
+      return res.data;
+    });
+  };
+
+  const selectedLeaderboard = useQuery(
+    ["selectedLeaderboard", leaderboardsQuery, selectedIndex],
```

Comment:
It should be leaderbirdsQuery.data I think. It’s for sure  If it works like useEffect